com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/RemoveNewAppsSuffixStep.java

```diff
@@ -0,0 +1,52 @@
+package com.sap.cloud.lm.sl.cf.process.steps;
+
+import com.sap.cloud.lm.sl.cf.core.model.ApplicationUpdateState;
+import com.sap.cloud.lm.sl.cf.process.Constants;
+import com.sap.cloud.lm.sl.cf.process.message.Messages;
+import org.cloudfoundry.client.lib.CloudControllerClient;
+import org.flowable.engine.delegate.DelegateExecution;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.annotation.Scope;
+
+import javax.inject.Named;
+import java.util.List;
+
+@Named("removeNewAppsSuffixStep")
+@Scope(BeanDefinition.SCOPE_PROTOTYPE)
+public class RemoveNewAppsSuffixStep extends SyncFlowableStep {
+
+    @Override
+    protected StepPhase executeStep(ExecutionWrapper execution) {
+        if (!shouldExecuteStep(execution.getContext())) {
+            return StepPhase.DONE;
+        }
+        getStepLogger().info(Messages.RENAMING_NEW_APPLICATIONS);
+
+        List<String> appsToProcess = StepsUtil.getAppsToDeploy(execution.getContext());
+        CloudControllerClient client = execution.getControllerClient();
+
+        for (String appName : appsToProcess) {
+            getStepLogger().debug(Messages.RENAMING_APPLICATION, appName);
+            client.rename(appName, removeAppNameSuffix(appName));
+        }
+        return StepPhase.DONE;
+    }
+
+
+    private boolean shouldExecuteStep(DelegateExecution context) {
```

Comment:
This method is duplicated. Extract it somewhere and reuse it.

Reply:
Extracted to StepsUtil